app/views/submissions/show.html.erb

```diff
@@ -0,0 +1,40 @@
+
+<div class="page-header" >
+  <h2><%= @submission.name %></h2>  <!-- but also title type header  -->
+</div>
+
+<script>pollForUpdates('<%= @project.id %>', '<%= @submission.id %>')</script>
+
+<%= link_to 'Back', project_path(@project), class: "btn btn-info" %><br><br>
+
+<h3>All the job results for <%= @submission.name%></h3>
+<% @submission.jobs.each do |job| %>
```

Comment:
A table is more appropriate here. The table can just show all the jobs submitted, and information for each job. No need for progressive disclosure via button click. In fact, a table showing everything would be okay even if we dumped the entire contents of the error and output files, as this won't be a common view accessed.

Reply:
Actually it would be better to just add “child rows” to the project “show” table. Then controls for stopping a job or providing links to the std error or output and status would go there. It simplifies the interface (this view goes away) and removes some code complexity (don’t have to get the “last job” anymore, just show them all) and doesn’t hide important information from the user, such as if they happen to have two jobs running for the same script which is definitely not desirable.

Reply:
>keeping that interface clean and simple makes sense

To clarify, I agree but realize that "clean and simple" depends on a lot of things. If a project contains 10 scripts and 1000 jobs it be nice to break up a single view into several like we have here. In that sense you see less, so its "simple". If a project contains 3 scripts and 6 jobs, it is simpler to have all the information on one page: you can see everything at a glance and you don't even have to scroll.

Based on the only use case we have been given, for a given project uploaded, there may be 1-3 scripts created, and for each 1-3 jobs, given that the two common error cases are exceeding walltime and using too many licenses, so you could imagine 3 jobs per script, one to adjust the walltime, another to address the license issue, one that works.

I guess the point I'm trying to make is it took a lot more code complexity to produce the current solution in anticipation of a workflow we have not yet observed or been told we need to support (having an uploaded project where you would have >5 scripts and >5 jobs per script). It is better in the future to build the minimum viable product that satisfies known requirements and wait and see before building more features.